sway/commands/focus.c

```diff
@@ -50,8 +50,11 @@ static struct cmd_results *focus_mode(struct sway_container *con,
 	} else {
 		new_focus = seat_get_focus_inactive_tiling(seat, ws);
 	}
-	if (!new_focus) {
-		new_focus = ws;
+	if (!new_focus || new_focus == ws ||
+			new_focus == ws->sway_workspace->floating) {
+		return cmd_results_new(CMD_FAILURE, "focus",
+				"Failed to find a %s container in workspace",
+				floating ? "floating" : "tiling");
```

Comment:
If there are no tiling containers and I'm changing focus from floating to tiled, shouldn't `focus == ws` be a valid thing?

Also the `seat_get_focus_inactive` functions shouldn't return the `L_FLOATING` container. If that's happening, we should find out why and fix it there.

Reply:
I don't really agree, but I've rebased it to allow it.